src/utils.c

```diff
@@ -424,3 +424,100 @@ SEXP startsWithAny(const SEXP x, const SEXP y, SEXP start) {
   return ScalarLogical(false);
 }
 
+SEXP frev(SEXP x, SEXP copyArg) {
+  if (INHERITS(x, char_dataframe) || INHERITS(x, char_datatable))
```

Comment:
In proper usage, `inherits('data.table')` should imply `inherits('data.frame')`, so I would drop the latter test. Users violating the package's own S3 setup cannot expect normal behavior.

Reply:
We have a similar check in assign.c
https://github.com/Rdatatable/data.table/blob/54f904831cafc37ba841958fbdbc614b8baac4c8/src/assign.c#L307-L309

Reply:
That's an internal error, so situation that should not happen